llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1924,6 +1924,16 @@ bool LoopVectorizationLegality::canFoldTailByMasking() const {
     }
   }
 
+  // The only loops we can vectorize without a scalar epilogue, are loops with
+  // a bottom-test and a single exiting block. We'd have to handle the fact
+  // that not every instruction executes on the last iteration.  This will
+  // require a lane mask which varies through the vector loop body.  (TODO)
+  if (TheLoop->getExitingBlock() != TheLoop->getLoopLatch()) {
```

Comment:
It might be worth moving this to the top of the function to avoid unnecessary work when we have multiple exiting blocks?

Reply:
Fixed. Thanks!